pkg/grpc/api.go

```diff
@@ -347,15 +348,20 @@ func (a *api) PublishEvent(ctx context.Context, in *runtimev1pb.PublishEventRequ
 	data := body
 
 	if !rawPayload {
-		envelope, err := runtimePubsub.NewCloudEvent(&runtimePubsub.CloudEvent{
+		cloudevent := runtimePubsub.CloudEvent{
 			ID:              a.id,
 			Topic:           in.Topic,
 			DataContentType: in.DataContentType,
 			Data:            body,
 			TraceID:         corID,
 			TraceState:      traceState,
 			Pubsub:          in.PubsubName,
-		})
+		}
+
+		// metadata beginning with "cloudevent-" are considered overrides to the cloudevent envelope
+		mapstructure.WeakDecode(in.Metadata, &cloudevent) // allows ignoring of case
```

Comment:
What if someone were to set properties like `data` or `pubsub` or `topic` in the metadata? Wouldn't that overwrite other properties we care about?

Reply:
They would have to do so as "cloudevent-data" or "cloudevent-topic" in metadata. The topic is actually stored in another place and additionally in the cloud event. This topic that can be overridden here is not used for routing.

This is by design.